core/domain/classifier_services.py

```diff
@@ -100,8 +101,32 @@ def classify_string_classifier_rule(state, normalized_answer):
             training_examples.extend([
                 [doc, [str(answer_group_index)]]
                 for doc in classifier_rule_spec.inputs['training_data']])
-    if len(training_examples) > 0:
+
+    exploration = exp_services.get_exploration_by_id(exp_id,
+                                                     version=exp_version)
+    algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput']
+    if state.classifier_model_id:
+        classifier = get_classifier_by_id(state.classifier_model_id)
+        sc_dict = classifier.cached_classifier_data
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'])
+        sc.from_dict(sc_dict)
+
+    else:
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'])
         sc.train(training_examples)
+        cached_classifier_data = sc.to_dict()
+        schema_version = exploration.states_schema_version
+        classifier = classifier_domain.Classifier('0', exploration.id,
+                                                  exp_version,
+                                                  state_name, algorithm_id,
+                                                  cached_classifier_data,
+                                                  schema_version)
+        classifier_id = save_classifier(classifier)
+        state.classifier_model_id = classifier_id
```

Comment:
This won't work actually. You'll have to update the state dict within the exploration and save the exploration. You might be able to find a similar example somewhere for sure.